runtime/src/snapshot_package.rs

```diff
@@ -223,52 +223,42 @@ pub struct SnapshotPackage {
 
 impl SnapshotPackage {
     pub fn new(accounts_package: AccountsPackage, accounts_hash: Hash) -> Self {
-        assert!(
-            accounts_package.snapshot_info.is_some(),
-            "The AccountsPackage must have snapshot info in order to make a SnapshotPackage!"
-        );
-        let snapshot_info = accounts_package.snapshot_info.unwrap();
+        let AccountsPackageType::Snapshot(snapshot_type) = accounts_package.package_type else {
+            panic!("The AccountsPackage must be of type Snapshot in order to make a SnapshotPackage!");
+        };
+        let Some(snapshot_info) = accounts_package.snapshot_info else {
```

Comment:
Non-blocking:

I think the old syntax here is maybe a bit more clean/concise. This is equivalent to:
```
let snapshot_info = accounts_package.snapshot_info.expect("...");
```
right?